hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/BaseRollbackActionExecutor.java

```diff
@@ -241,7 +247,9 @@ public List<HoodieRollbackStat> doRollbackAndGetStats(HoodieRollbackPlan hoodieR
    * @return list of {@link HoodieRollbackStat}s.
    */
   protected List<HoodieRollbackStat> executeRollback(HoodieInstant instantToRollback, HoodieRollbackPlan rollbackPlan) {
-    return new BaseRollbackHelper(table.getMetaClient(), config).performRollback(context, instantToRollback, rollbackPlan.getRollbackRequests());
+    return table.version().lesserThan(HoodieTableVersion.EIGHT)
+        ? new BaseRollbackHelperTableVersionSix(table, config).performRollback(context, instantTime, instantToRollback, rollbackPlan.getRollbackRequests())
+        : new BaseRollbackHelper(table.getMetaClient(), config).performRollback(context, instantToRollback, rollbackPlan.getRollbackRequests());
```

Comment:
Could we fold the table version check into `BaseRollbackHelper` instead of here?

Reply:
We did attempt that at first, but there are 3 to 4 `if else` blocks. and we felt this way, we can get rid of the code easily. rather than having lot of `if else` blocks. and any new fixes should only target BaseRollbackHelper class.